platforms/native/language-native/src/integTest/groovy/org/gradle/language/AbstractNativeLanguageIncrementalBuildIntegrationTest.groovy

```diff
@@ -386,7 +386,7 @@ abstract class AbstractNativeLanguageIncrementalBuildIntegrationTest extends Abs
         def snapshot = executable.snapshot()
 
         and:
-        def linkerArgs = toolChain.isVisualCpp() ? "'/DEBUG'" : OperatingSystem.current().isMacOsX() ? "'-Xlinker', '-no_pie'" : "'-Xlinker', '-q'"
+        def linkerArgs = toolChain.isVisualCpp() ? "'/DEBUG'" : OperatingSystem.current().isMacOsX() ? "'-Xlinker', '-O0'" : "'-Xlinker', '-q'"
```

Comment:
`-no_pie` has no effect on apple silicon machines. Use a different flag that will generate a binary with different content on both macOS machines.

Reply:
🤔 Might be good to extract that flag in a variable that is named explicitly or has a comment indicating what its purpose is.

Reply:
Done